plugins/action/horizon_renew.py

```diff
@@ -16,26 +16,50 @@ class ActionModule(HorizonAction):
     TRANSFERS_FILES = True
 
     def _args(self):
-        return ['certificate_id', 'certificate_pem', 'password', 'csr', 'private_key']
+        return ['certificate_id', 'certificate_pem', 'password', 'csr', 'private_key', 'mode']
     
     def run(self, tmp=None, task_vars=None):
         result = super(ActionModule, self).run(tmp, task_vars)
 
         try: 
             client = self._get_client()
             content = self._get_content()
+
+            if "mode" in content:
+                should_generate_csr = content["mode"] == "decentralized" and content['csr'] is None and content["private_key"] is not None
```

Comment:
Un peu dure à lire cette ligne, how about 
```suggestion
            if "mode" in content:
                is_decentralized = content["mode"] == "decentralized"
                has_no_csr = content.get('csr') is None
                has_private_key = content.get("private_key") is not None

                should_generate_csr = is_decentralized and has_no_csr and has_private_key
```

Reply:
@AdrienDucourthial wdyt?

Reply:
looks like a good idea